### PR TITLE
Add new endpoint to delete users

### PR DESCRIPTION
**Pull request by jfcalvo**
Add new endpoint to delete users.
- The endpoint returns a 404 error when the provided `user_id` is not found.
- I have found a problem with `UUID` type not being correctly serialized to JSON for some reason. We need to find an alternative solution to this (more info in the code).

### CODE REVIEW

File: src/argilla/server/apis/v0/handlers/users.py [hunk 1]
```diff
@@ -54,3 +57,20 @@ def create_user(
...
+def delete_user(
+    *, db: Session = Depends(get_db), user_id: UUID, current_user: User = Security(auth.get_user, scopes=[])
+):
+    user = accounts.get_user_by_id(db, user_id)
+    if not user:
+        # TODO: Forcing here user_id to be an string.
+        # Not casting it is causing a `Object of type UUID is not JSON serializable`.
+        # Possible solution redefining JSONEncoder.default here:
+        # https://github.com/jazzband/django-push-notifications/issues/586
+        raise EntityNotFoundError(name=str(user_id), type=User)
```
Comment (frascuchon) [inline src/argilla/server/apis/v0/handlers/users.py:81]: I think adding `json_encoders`  to out error class models should work.

Take a look here: https://docs.pydantic.dev/usage/exporting_models/#json_encoders

File: src/argilla/server/apis/v0/handlers/users.py [hunk 2]
```diff
@@ -54,3 +57,20 @@ def create_user(
...
+
+    accounts.delete_user(db, user)
+
+    return User.from_orm(user)
```
Comment (frascuchon) [inline src/argilla/server/apis/v0/handlers/users.py:85]: Here, do you have any problem with UUID?

File: src/argilla/server/contexts/accounts.py [hunk 3]
```diff
@@ -43,6 +49,13 @@ def create_user(db: Session, user_create: UserCreate):
     return user
 
 
+def delete_user(db: Session, user: User):
+    db.delete(user)
```
Comment (frascuchon) [inline src/argilla/server/contexts/accounts.py:57]: Removing this user from the user_workspaces relationship table will be tackled in a next step, right?

File: src/argilla/server/apis/v0/handlers/users.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/apis/v0/handlers/users.py:81]: I will take a look.

File: src/argilla/server/apis/v0/handlers/users.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla/server/apis/v0/handlers/users.py:85]: Nope, on Pydantic schemas UUID it's correctly serialized. I'm using it with workspaces without any problem.

File: src/argilla/server/contexts/accounts.py (hunk 3 above)
Comment (jfcalvo) [reply src/argilla/server/contexts/accounts.py:57]: That's correct.

File: src/argilla/server/apis/v0/handlers/users.py (hunk 1 above)
Comment (frascuchon) [reply src/argilla/server/apis/v0/handlers/users.py:81]: We will resolve this once we review the error classes

**Review by frascuchon — APPROVED** [review]

**Pull request merged**